compiler/src/iree/compiler/Codegen/Common/test/link_tuning_specs.mlir

```diff
@@ -135,3 +135,208 @@ module @td_module_4 attributes { transform.with_named_sequence } {
+        }
+    }
+}
+
+// CHECK-LABEL:   module @td_module
+// CHECK-SAME:    iree_codegen.tuning_spec_with_default_entrypoint
+// CHECK-NOT:     @mmt_module
+// CHECK-NOT:     @attention_module
+// CHECK:         @__kernel_config(
+// CHECK:         transform.foreach_match
+// CHECK:         @match_mmt_f16_f16_f32 -> @apply_op_config
+// CHECK-NEXT:    @match_attention_f16 -> @apply_attn_op_config
+// CHECK-NEXT:    transform.yield
+
+// -----
+
+// Make sure that all named sequence operation names in the merged foreach_match remain unique.
+
+module @td_module attributes { transform.with_named_sequence } {
+   module @inner_module_a
+    attributes { transform.with_named_sequence, iree_codegen.tuning_spec_with_default_entrypoint } {
+    transform.named_sequence @match(%arg: !transform.any_op {transform.readonly}) -> (!transform.any_op) {
+      transform.yield %arg : !transform.any_op
+    }
+
+    transform.named_sequence @apply_op_config(%op: !transform.any_op {transform.readonly}) {
+      transform.yield
+    }
+
+    transform.named_sequence @__kernel_config(%arg0: !transform.any_op {transform.consumed})
+      -> (!transform.any_op) attributes { iree_codegen.tuning_spec_entrypoint } {
+      %res = transform.foreach_match in %arg0 @match -> @apply_op_config
+        : (!transform.any_op) -> (!transform.any_op)
+      transform.yield %res : !transform.any_op
+    }
+  }
+
+  module @inner_module_b
+    attributes { transform.with_named_sequence, iree_codegen.tuning_spec_with_default_entrypoint } {
+    transform.named_sequence @match(%arg: !transform.any_op {transform.readonly}) -> (!transform.any_op) {
+      transform.yield %arg : !transform.any_op
+    }
+
+    transform.named_sequence @apply_op_config(%op: !transform.any_op {transform.readonly}) {
+      transform.yield
+    }
+
+    transform.named_sequence @__kernel_config(%arg0: !transform.any_op {transform.consumed})
+      -> (!transform.any_op) attributes { iree_codegen.tuning_spec_entrypoint } {
+      %res = transform.foreach_match in %arg0 @match -> @apply_op_config
+        : (!transform.any_op) -> (!transform.any_op)
+      transform.yield %res : !transform.any_op
+    }
+  }
+
+  module @inner_module_c
+    attributes { transform.with_named_sequence, iree_codegen.tuning_spec_with_default_entrypoint } {
+    transform.named_sequence @match(%arg: !transform.any_op {transform.readonly}) -> (!transform.any_op) {
+      transform.yield %arg : !transform.any_op
+    }
+
+    transform.named_sequence @apply_op_config_1(%op: !transform.any_op {transform.readonly}) {
+      transform.yield
+    }
+
+    transform.named_sequence @__kernel_config(%arg0: !transform.any_op {transform.consumed})
+      -> (!transform.any_op) attributes { iree_codegen.tuning_spec_entrypoint } {
+      %res = transform.foreach_match in %arg0 @match -> @apply_op_config_1
+        : (!transform.any_op) -> (!transform.any_op)
+      transform.yield %res : !transform.any_op
+    }
```

Comment:
Can you update this test and change the order of named sequences? match doesn't have to be the first op, while kernel_config doesn't always have to be the last one IIUC.